make/CompileInterimLangtools.gmk

```diff
@@ -45,8 +45,21 @@ $(BUILDTOOLS_OUTPUTDIR)/gensrc/%.interim/module-info.java: \
 	$(call MakeDir, $(@D))
 	$(SED) $(INTERIM_MODULEINFO_PATTERN) $< > $@
 
+INTERIM_TOOL_PROVIDER_PATTERN := \
+    $(foreach m, $(INTERIM_LANGTOOLS_BASE_MODULES), -e 's/"$m"/"$m.interim"/g')
+
+$(BUILDTOOLS_OUTPUTDIR)/gensrc/java.compiler.interim/javax/tools/ToolProvider.java: \
+    $(TOPDIR)/src/java.compiler/share/classes/javax/tools/ToolProvider.java
+	$(call LogInfo, Generating ToolProvider.java for java.compiler.interim)
+	$(call MakeDir, $(@D))
+	$(SED) $(INTERIM_TOOL_PROVIDER_PATTERN) $< > $@
+
```

Comment:
Could you introduce a new section here with a comment header briefly describing what this is doing and why? It's currently under a section describing module-info.java. Let the last line of each section be the `TARGETS += ...` line.